pkg/operator/sync.go

```diff
@@ -1465,6 +1470,61 @@ func (optr *Operator) getCAsFromConfigMap(namespace, name, key string) ([]byte,
 	return getCAsFromConfigMap(cm, key)
 }
 
+// This function stamps the current operator version and commit hash in the boot images configmap
+// that lives in the MCO namespace. Before doing so, it ensures that the pool is targeting an MC
+// that is generated by the current version of the MCO and that the pool atleast has 1 upgraded node(or
+// has completed an upgrade). This "stamp" is used by the machine set controller as a safety before
+// it updates boot images.
+
+func (optr *Operator) stampBootImagesCM(pool *mcfgv1.MachineConfigPool) error {
```

Comment:
this is basically the same logic as the MCS served config, right?

Reply:
Yep, the only difference being, here we gate it on 1 master node completing the update, while the MCS wait for at least 1 node in the pool to complete the update before serving it to new nodes.